### ISSUE TITLE
ResourceProfiler raises NoSuchProcess exception

### ISSUE THREAD

**Issue by pwaller** [dask/dask#2284]
I have hit this bug reliably on 0.11 and 0.14.1.

I am doing the following, for a moderately complicated graph, on a 16 core machine. The profiling code is taken straight from the dask documentation.

```python
from dask.diagnostics import Profiler, ResourceProfiler, CacheProfiler, visualize
from cachey import nbytes

with Profiler() as prof, ResourceProfiler(dt=0.25) as rprof, CacheProfiler(metric=nbytes) as cprof:
    x.compute()
```

It reliably gives the following error:

```pytb
Process _Tracker-50:
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=69516)
Traceback (most recent call last):
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/_pslinux.py", line 1332, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/_pslinux.py", line 1528, in memory_info
    with open_binary("%s/%s/statm" % (self._procfs_path, self.pid)) as f:
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/_pslinux.py", line 190, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/69516/statm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/home/pwaller/.local/lib/python3.5/site-packages/dask/diagnostics/profile.py", line 231, in run
    mem = sum(p.memory_info().rss for p in ps) / 1e6
  File "/home/pwaller/.local/lib/python3.5/site-packages/dask/diagnostics/profile.py", line 231, in <genexpr>
    mem = sum(p.memory_info().rss for p in ps) / 1e6
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/_common.py", line 313, in wrapper
    return fun(self)
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/__init__.py", line 1108, in memory_info
    return self._proc.memory_info()
  File "/home/pwaller/.local/lib/python3.5/site-packages/psutil/_pslinux.py", line 1338, in wrapper
    raise NoSuchProcess(self.pid, self._name)
```

I assume the problem here is that the multiprocessing code is waiting on the process before `memory_info()` is called, so the process has gone away by the time the profiler asks for the memory usage.

**Comment by jcrist** [comment]
Thanks for the bug report. Does this error happen early on in the computation, or towards the end? I can see how we could have a race condition in shutdown of the process tracker, but not in startup, just want to make sure that's what you're seeing.

**Comment by pwaller** [comment]
Judging from the time taken for the error to appear, it's at the end, but it's difficult to be certain. It seems likely that it's a race at the end, given the nature of the bug.

**Comment by jcrist** [comment]
This should be fixed in #2284.

**Comment by Rblivingstone** [comment]
I'm sure if this is a new problem, or this old issue coming up again. I was running a fairly sizeable computation and I got the following error:

```
Traceback (most recent call last):
  File "C:\Users\rbarnes\AppData\Local\Continuum\Anaconda3\lib\site-packages\psutil\_pswindows.py", line 536, in wrapper
    return fun(self, *args, **kwargs)
  File "C:\Users\rbarnes\AppData\Local\Continuum\Anaconda3\lib\site-packages\psutil\_pswindows.py", line 680, in create_time
    return cext.proc_create_time(self.pid)
ProcessLookupError: [Errno 3] No such process

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\rbarnes\AppData\Local\Continuum\Anaconda3\lib\site-packages\psutil\__init__.py", line 390, in _init
    self.create_time()
  File "C:\Users\rbarnes\AppData\Local\Continuum\Anaconda3\lib\site-packages\psutil\__init__.py", line 655, in create_time
    self._create_time = self._proc.create_time()
  File "C:\Users\rbarnes\AppData\Local\Continuum\Anaconda3\lib\site-packages\psutil\_pswindows.py", line 541, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=10424)
```

I'm running dask 0.16.1 on windows machine with 8 cores.

I was essentially doing this:

```
groups = df.groupby(['ApplicationId']).max()
mem = df[df['index'].isin(groups['index'])]
```
